### ISSUE TITLE
Document Twisted Names examples

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5588]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)| @ashfall reported|
|-|-|
|Trac ID|trac#5588|
|Type|task|
|Created|2012-04-01 16:35:47Z|

Documentation on how to run the Twisted Names examples at http://twistedmatrix.com/documents/current/names/examples/index.html
and description for each of the scripts should be added.

Something like the description at:
http://twistedmatrix.com/documents/current/web/examples/index.html

Attachments:

* [5588-names-examples.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5fd/5fd36db6e87d59d8f56a9f111203f627d53bec8e/b7b560f65c6707b906691504cf3659e1d21a6483.patch) (2820 bytes) - added by ashfall on 2012-04-01 18:47:28Z - Added description.
* [5588-topfile.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5fd/5fd36db6e87d59d8f56a9f111203f627d53bec8e/0f610aee03acce78dfe0e57378b8df1f91436a1e.patch) (331 bytes) - added by ashfall on 2012-04-01 20:24:22Z - Added topfile.
* [5588-topfile.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5fd/5fd36db6e87d59d8f56a9f111203f627d53bec8e/569d7183830bad7a3a738c9f70ca1fb43a2363c1.patch) (317 bytes) - added by ashfall on 2012-04-02 15:01:32Z - New topfile as per exarkun's review.

<details><summary>Searchable metadata</summary>

```
trac-id__5588 5588
type__task task
reporter__ashfall ashfall
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__names names
keywords__documentation documentation
time__1333298147000000 1333298147000000
changetime__1333380418000000 1333380418000000
version__None None
owner__exarkun exarkun
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)|@ashfall set owner to @ashfall|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)|@ashfall removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Thanks!  These changes look really good.

The only thing I would do differently has to do the news file.  I think it's probably sufficient to say _The examples_ rather than _The Twisted Names examples_, since this will end up in either [tags/releases/twisted-12.0.0/twisted/names/topfiles/NEWS the Twisted Names news file](tags/releases/twisted-12.0.0/twisted/names/topfiles/NEWS the Twisted Names news file) or [tags/releases/twisted-12.0.0/NEWS#L93 the Twisted Names section of the overall news file](tags/releases/twisted-12.0.0/NEWS#L93 the Twisted Names section of the overall news file).

This is a very minor change (not to mention a fairly subtle stylistic one), so I'm happy to make it myself when applying your patch.  Thanks again!

One other thought comes to mind as I look at these examples, though.  They don't do a _great_ job of demonstrating the functionality they're supposed to demonstrate.  In particular, they generate rather opaque, uninformative output.  I've filed [ticket:5594 a new ticket for improving their output].

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [34056]) Add descriptions and run instructions for Twisted Names examples

Author: ashfall
Reviewer: exarkun
Fixes: [#5588](https://github.com/twisted/twisted/issues/5588)

Each Twisted Names example now starts with instructions in how to run it, as well as
a description of what it is supposed to demonstrate.
